tests/API/Management/ClientsTest.php

```diff
@@ -2,42 +2,197 @@
 
-    protected function getApiClient() {
-        $env = $this->getEnv();
-        $token = $this->getToken($env, [
-            'clients' => [
-                'actions' => ['create', 'read', 'delete', 'update']
-            ]
-        ]);
+    /**
+     * Name of the updated Client.
+     * Appended with a random string in self::__construct().
+     *
+     * @var string
+     */
+    protected $update_client_name = 'TEST-UPDATE-CLIENT-';
 
-        $this->domain = $env['DOMAIN'];
+    /**
+     * Application type of the updated Client.
+     *
+     * @var string
+     */
+    protected $update_app_type = Clients::APP_TYPE_NON_INTERACTIVE;
 
-        $api = new Management($token, $env['DOMAIN']);
+    /**
+     * SSO setting of the updated Client.
+     *
+     * @var bool
+     */
+    protected $update_sso = true;
 
+    /**
+     * Description of the updated Client.
+     *
+     * @var string
+     */
+    protected $update_desc = '__Auth0_PHP_updated_app_description__';
+
+    /**
+     * ClientsTest constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->create_client_name .= rand();
+        $this->update_client_name .= rand();
+    }
+
+    /**
+     * Return the ID for the entity created.
+     *
+     * @param array $entity - Client created during test.
+     *
+     * @return string|integer
+     */
+    protected function getId($entity)
+    {
+        return $entity[$this->id_name];
+    }
+
+    /**
+     * Return the Clients API to test.
+     *
+     * @return Management\Clients
+     */
+    protected function getApiClient()
+    {
+        $token = $this->getToken($this->env, [ 'clients' => [ 'actions' => ['create', 'read', 'delete', 'update' ] ] ]);
+        $api = new Management($token, $this->domain);
         return $api->clients;
     }
 
-    protected function getCreateBody() {
-        $client_name = 'test-create-client' . rand();
-        echo "\n-- Using client name $client_name \n";
+    /**
+     * Get the Client create data to send with the test create call.
+     *
+     * @return array
+     */
+    protected function getCreateBody()
+    {
+        return [
+            'name' => $this->create_client_name,
+            'app_type' => $this->create_app_type,
+            'sso' => $this->create_sso,
+            'description' => $this->create_desc,
+        ];
+    }
+
+    /**
+     * Tests the \Auth0\SDK\API\Management\Clients::getAll() method.
+     *
+     * @param Management\Clients $client - API client to use.
+     * @param array $created_entity - Entity created during create() test.
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    protected function getAll($client, $created_entity)
+    {
+        $fields = array_keys($this->getCreateBody());
+        $fields[] = $this->id_name;
+
+        // Check that pagination works.
+        $all_results = $client->getAll($fields, true, $this->create_app_type, 1, 1);
```

Comment:
this is getting page 1 (of 0 based) and 1 result per page. If a single client exists, that one would belong to page 0 right? So this is returning the second client. Is that assumption OK ?